Tactics/SymContext.lean

```diff
@@ -161,7 +197,16 @@ def fromLocalContext (state? : Option Name) : MetaM SymContext := do
   let h_run ← findLocalDeclUsernameOfTypeOrError h_run_type
 
   -- Unwrap and reflect `runSteps`
-  let runSteps ← withErrorContext h_run h_run_type <| reflectNatLiteral runSteps
+  let runSteps? ← do
+    let msg := m!"Reflecting: {runSteps}"
+    withTraceNode `Tactic.sym (fun _ => pure msg) <| do
+      try
+        let runSteps ← reflectNatLiteral runSteps
```

Comment:
Check my understanding please: `fromLocalContext` is invoked after the `sym_while`, which is why at this point we can expect that `runSteps` can be reflected internally as a `Nat`. Right?

Reply:
No, `fromLocalContext` is invoked only once, at the very beginning of `sym_n`. Here, the `reflectNatLiteral` is fine because it's wrapped in a try-catch, which will set the field to `none` if reflection fails. (I should really be using `reflectNatLiteral?` here, which I introduced because of this very callsite, I should fix that)

Reply:
Thanks, @alexkeizer. That helps!